llvm/lib/Target/AIE/AIE2RegisterInfo.cpp

```diff
@@ -457,3 +457,33 @@ AIE2RegisterInfo::getCoveringSubRegs(const TargetRegisterClass &RC) const {
   }
   return Subregs;
 }
+
+bool AIE2RegisterInfo::isVecOrAccRegClass(const TargetRegisterClass *RC) const {
```

Comment:
Hi @F-Stuckmann, as we already encode vector types tied to register classes, I suggest a very compact implementation as follows:
```
bool AIE2RegisterInfo::isVecOrAccRegClass(const TargetRegisterClass *RC) const {
  // ******** Vector classes ********
  for (auto I = legalclasstypes_begin(*RC); I != legalclasstypes_end(*RC);
       ++I) {
    EVT VT = *I;
    if (VT.isVector())
      return true;
  }

  return false;
}

This can be also moved to the base class. In this case we also avoid target-specific knowledge. 
```

Reply:
accepted the suggestion and moved the method to AIEBaseRegisterinfo.h

Reply:
I did not accept this change, because MVT are used earlier in the pipeline and in the register renaming pass I have access to the actual registers, which i can querry with my implementation.